test/unit/modules/blocks/utils.js

```diff
@@ -13,268 +13,581 @@
+
+var viewRow_full_blocks_list = [
+	{
+		b_id: '13068833527549895884',
+		b_height: 3,
+		t_id: '6950874693022090568',
+		t_type: 0,
+	},
+];
+
 describe('blocks/utils', () => {
-	describe('Utils', () => {
-		describe('library', () => {
-			it('should assign logger');
+	var dbStub;
+	var loggerStub;
+	var blockMock;
+	var transactionMock;
+	var accountMock;
+	var blocksUtilsModule;
+	var modulesStub;
+	var __private;
+	var library;
+	var modules;
+
+	beforeEach(() => {
+		dbStub = {
+			blocks: {
+				getIdSequence: sinonSandbox.stub().resolves(),
+				getHeightByLastId: sinonSandbox.stub().resolves(['1']),
+				loadLastBlock: sinonSandbox.stub().resolves(viewRow_full_blocks_list),
+				loadBlocksData: sinonSandbox.stub(),
+				aggregateBlocksReward: sinonSandbox.stub().resolves(),
+			},
+		};
+
+		dbStub.blocks.loadBlocksData
+			.withArgs(sinonSandbox.match({ id: '13068833527549895884' }))
+			.resolves(viewRow_full_blocks_list)
+			.withArgs(sinonSandbox.match({ id: '1' }))
+			.resolves([]);
+
+		loggerStub = {
+			trace: sinonSandbox.spy(),
+			info: sinonSandbox.spy(),
+			error: sinonSandbox.spy(),
+		};
+
+		blockMock = {
+			dbRead: function(input) {
+				return { id: input.b_id, height: input.b_height };
+			},
+		};
+
+		transactionMock = {
+			dbRead: function(input) {
+				return { id: input.t_id, type: input.t_type };
+			},
+		};
+
+		accountMock = {
+			get: sinonSandbox.stub(),
+		};
+
+		accountMock.get
+			.withArgs(sinonSandbox.match({ address: 'ERRL' }))
+			.callsArgWith(1, 'Address error stub', null)
+			.withArgs(sinonSandbox.match({ address: '0L' }))
+			.callsArgWith(1, null, undefined)
+			.withArgs(sinonSandbox.match({ address: '1L' }))
+			.callsArgWith(1, null, { publicKey: '123abc' });
+
+		blocksUtilsModule = new BlocksUtils(
+			loggerStub,
+			accountMock,
+			blockMock,
+			transactionMock,
+			dbStub,
+			modulesLoader.scope.dbSequence,
+			modulesLoader.scope.genesisblock
+		);
 
-			it('should assign db');
+		modulesStub = {
+			blocks: {
+				lastBlock: {
+					get: sinonSandbox
+						.stub()
+						.returns({ id: '9314232245035524467', height: 1 }),
+					set: sinonSandbox
+						.stub()
+						.returns({ id: '9314232245035524467', height: 1 }),
+				},
+				utils: {
+					readDbRows: blocksUtilsModule.readDbRows,
+				},
+			},
+		};
+
+		blocksUtilsModule.onBind(modulesStub);
+
+		__private = BlocksUtils.__get__('__private');
+		__private.loaded = false;
+
+		library = BlocksUtils.__get__('library');
+		modules = BlocksUtils.__get__('modules');
+	});
 
-			it('should assign dbSequence');
+	afterEach(() => {
+		sinonSandbox.reset();
+	});
 
-			it('should assign genesisblock');
+	describe('constructor', () => {
+		describe('library', () => {
+			it('should assign logger', () => {
+				expect(library.logger).to.eql(loggerStub);
+			});
 
 			describe('should assign logic', () => {
-				it('should assign block');
+				it('should assign account', () => {
```

Comment:
beforeEach will trigger before this it - we have entire new instance here

Reply:
As above.